pkg/operator/ceph/object/cosi/controller.go

```diff
@@ -0,0 +1,291 @@
+	if err != nil {
+		return reconcile.Result{}, *cephCOSIDriver, fmt.Errorf("failed to list object stores. %+v", err)
+	}
+
+	// Check for object store part of Multisite
+	if len(list.Items) == 0 {
+		return reconcile.Result{}, *cephCOSIDriver, fmt.Errorf("no object store found, requeuing")
+	}
+
+	// TODO create CephObjectStoreUser name cosi for each object store, if not already created
+
+	// Start the Ceph COSI Driver
+	err = r.startCephCOSIDriver(cephCOSIDriver)
+	if err != nil {
+		return reconcile.Result{}, *cephCOSIDriver, fmt.Errorf("failed to start Ceph COSI Driver. %+v", err)
+	}
+
+	return reconcile.Result{}, *cephCOSIDriver, nil
+}
+
+// Start the Ceph COSI Driver
+func (r *ReconcileCephCOSIDriver) startCephCOSIDriver(cephCOSIDriver *cephv1.CephCOSIDriver) error {
+	// Create the Ceph COSI Driver Deployment
+	cephcosidriverDeployment, err := createCephCOSIDriverDeployment(cephCOSIDriver)
+	if err != nil {
+		return fmt.Errorf("failed to create Ceph COSI Driver deployment. %+v", err)
+	}
+
+	_, err = r.context.Clientset.AppsV1().Deployments(cephCOSIDriver.Namespace).Create(r.opManagerContext, cephcosidriverDeployment, metav1.CreateOptions{})
+	if err != nil {
+		if !errors.IsAlreadyExists(err) {
+			// TODO updates
+			return fmt.Errorf("failed to create Ceph COSI Driver deployment. %+v", err)
+		}
+		logger.Infof("Ceph COSI Driver deployment already exists")
+	} else {
+		logger.Infof("Ceph COSI Driver deployment started")
+	}
+
+	return nil
+}
+
+func createCephCOSIDriverDeployment(cephCOSIDriver *cephv1.CephCOSIDriver) (*appsv1.Deployment, error) {
+	cosiPodSpec, err := createCOSIPodSpec(cephCOSIDriver)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create COSI pod spec. %+v", err)
+	}
+	strategy := appsv1.DeploymentStrategy{
+		Type: appsv1.RecreateDeploymentStrategyType,
+		RollingUpdate: &appsv1.RollingUpdateDeployment{
+			MaxSurge:       &intstr.IntOrString{IntVal: 1},
+			MaxUnavailable: &intstr.IntOrString{IntVal: 0},
+		},
+	}
+
+	replica := int32(1)
+	minReadySeconds := int32(30)
+	progressDeadlineSeconds := int32(600)
+	revisionHistoryLimit := int32(3)
+
+	cephcosidriverDeployment := &appsv1.Deployment{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      cephCOSIDriver.Name,
+			Namespace: cephCOSIDriver.Namespace,
+			Labels:    getCOSILabels(),
+		},
+		Spec: appsv1.DeploymentSpec{
+			Replicas: &replica,
+			Selector: &metav1.LabelSelector{
+				MatchLabels: getCOSILabels(),
+			},
+			Template:                cosiPodSpec,
+			Strategy:                strategy,
+			MinReadySeconds:         minReadySeconds,
+			ProgressDeadlineSeconds: &progressDeadlineSeconds,
+			RevisionHistoryLimit:    &revisionHistoryLimit,
+		},
+	}
+
+	return cephcosidriverDeployment, nil
+}
+
+func getCOSILabels() map[string]string {
+	// TODO Add Rook related common labels??
+	return map[string]string{
+		"app":                         "rook-ceph-cosi-driver",
```

Comment:
Thought: Following the CSI trend, this would be "ceph-cosi-driver" without "rook-"